Robinhood/Robinhood.py

```diff
@@ -1341,38 +1341,55 @@ def submit_order(self,
 
     def cancel_order(
             self,
-            order_id
-    ): 
+            order_id):
         """
-        Cancels specified order and returns the response (results from `orders` command). 
+        Cancels specified order and returns the response (results from `orders` command).
         If order cannot be cancelled, `None` is returned.
-
         Args:
-            order_id (str): Order ID that is to be cancelled or order dict returned from
+            order_id (str or dict): Order ID string that is to be cancelled or open order dict returned from
             order get.
         Returns:
             (:obj:`requests.request`): result from `orders` put command
         """
-        if order_id is str:
+        if isinstance(order_id, str):
             try:
-                order = self.session.get(self.endpoints['orders'] + order_id, timeout=15).json()
+                order = self.session.get(endpoints.orders() + order_id, timeout=15).json()
             except (requests.exceptions.HTTPError) as err_msg:
                 raise ValueError('Failed to get Order for ID: ' + order_id
                     + '\n Error message: '+ repr(err_msg))
-        else:
-            raise ValueError('Cancelling orders requires a valid order_id string')
 
-        if order.get('cancel') is not None:
-            try: 
-                res = self.session.post(order['cancel'], timeout=15)
-                res.raise_for_status()
+            if order.get('cancel') is not None:
+                try:
+                    res = self.session.post(order['cancel'], timeout=15)
+                    res.raise_for_status()
+                    return res
+                except (requests.exceptions.HTTPError) as err_msg:
+                    raise ValueError('Failed to cancel order ID: ' + order_id
+                         + '\n Error message: '+ repr(err_msg))
+                    return None
+
+        if isinstance(order_id, dict):
+            order_id = order_id['id']
+            try:
+                order = self.session.get(endpoints.orders() + order_id, timeout=15).json()
             except (requests.exceptions.HTTPError) as err_msg:
-                raise ValueError('Failed to cancel order ID: ' + order_id
-                     + '\n Error message: '+ repr(err_msg))
-                return None
-            
+                raise ValueError('Failed to get Order for ID: ' + order_id
+                    + '\n Error message: '+ repr(err_msg))
+
+            if order.get('cancel') is not None:
+                try:
+                    res = self.session.post(order['cancel'], timeout=15)
+                    res.raise_for_status()
+                    return res
+                except (requests.exceptions.HTTPError) as err_msg:
+                    raise ValueError('Failed to cancel order ID: ' + order_id
+                         + '\n Error message: '+ repr(err_msg))
+                    return None
+
+        elif order_id is not str or dict:
```

Comment:
Shouldn't this line be using `isinstance` like the rest if the if statement?

Like ```elif not isinstance(order_id,str) and not isinstance(order_id,dict):```

Reply:
Also might make sense to change your commit message to be more specific like "fix cancel_order method" since that is the message that will appear in the repo for the entire Robinhood.py file.